service/storage/storage.go

```diff
@@ -142,8 +142,34 @@ func (s *service) FileGetVersion(ctx context.Context, bucketID, fileID, version
 	return rc, fd, err
 }
 
-func (s *service) FileListVersions(ctx context.Context, bucketID, fileID string) ([]*models.FileDescriptor, error) {
-	return s.s3.List(ctx, bucketID, fileID)
+func (s *service) FileListVersions(ctx context.Context, bucketID, fileID string, createdAtSince, createdAtUntil *strfmt.DateTime) ([]*models.FileDescriptor, error) {
+	// init list to return
+	list := []*models.FileDescriptor{}
+
+	// check if bucket exists
+	exists, err := s.s3.BucketExists(ctx, bucketID)
+	if err != nil {
+		s.logger.Info().Err(err).Str("bucket", bucketID).Msg("Failed to check if bucket exists")
+		return nil, err
+	}
+	if !exists {
+		return list, nil
+	}
+
+	l, err := s.s3.List(ctx, bucketID, fileID)
+	if (createdAtSince == nil && createdAtUntil == nil) || err != nil {
+		return l, err
+	}
+
+	// extract only versions fitting the created timestamp filtering specified
+	for _, f := range l {
+		if (createdAtSince == nil || (createdAtSince != nil && time.Time(*createdAtSince).Before(time.Time(f.Created)))) &&
```

Comment:
invalid operation: f (variable of type *invalid type) has no field or method Created